src/components/tx-flow/flows/SignMessage/SignMessage.tsx

```diff
@@ -322,14 +324,14 @@ const SignMessage = ({ message, safeAppId, requestId }: ProposeProps | ConfirmPr
       ) : (
         <>
           <TxCard>
-            <AlreadySignedByOwnerMessage hasSigned={hasSigned} />
+            <AlreadySignedByOwnerMessage hasSigned={signedByCurrentSafe} />
 
             <InfoBox
               title="Collect all the confirmations"
               message={
-                requestId
+                requestId && !hasSignature
                   ? 'Please keep this modal open until all signers confirm this message. Closing the modal will abort the signing request.'
-                  : 'The signature will be submitted to the Safe App when the message is fully signed.'
+                  : 'The signature will be submitted to the App when the message is fully signed.'
```

Comment:
```suggestion
                  : 'The signature will be submitted to the requesting app when the message is fully signed.'
```

Technically, `requestId` is there only for Safe App messages. Swaps also, but they don't have messages (yet).